### ISSUE TITLE
Bug in "multiple" data_type handling

### ISSUE THREAD

**Issue by dongbohu** [greenelab/cimr#39]
Given a yaml file whose `data_type` is `multiple`, the downloaded file was not de-compressed correctly. Suppose the tarball's content is like this:
```
gwas
└── gwas.txt.gz
whatever
└── eqtl.txt.gz
```
But `cimr` doesn't report the error of unknown data_type `whatever`.

The correct behavior should be:
1. Check the subdirectory's name to verify that it is a valid name (`gwas`, `eqtl`, etc); if not, report the error and terminate the process immediately
2. De-archive the tarball file

**Comment by dongbohu** [comment]
`verify_dir()` located in https://github.com/greenelab/cimr/blob/592e67a17b6983563ac6e386ff1ad102c142baf1/cimr/processor/yamler.py#L187 should be changed to:
```python
def verify_dir(tarred_data):
    """Check directory tree of tarball containing multiple files"""
    for member in tarred_data.getmembers():
        if not member.name.startswith(DATA_TYPES):
            logging.error(f' data_type not indicated in dir tree.')
            sys.exit(1)
```

**Comment by ypar** [comment]
DATA_TYPES is a tuple, which is supported by str.startswith. I will take a look at this.

**Comment by dongbohu** [comment]
I didn't know that `str.startswith()` also accepts a tuple as argument, but it is still not good enough. What if the sub-dir in the archive is `gwas_foobar`? There are many corner cases in a tarball file's hierarchy. I will try to file a PR to fix this today.

**Comment by dongbohu** [comment]
Fixed in PR #40.
